src/modes/coincident_select.js

```diff
@@ -83,13 +83,30 @@ CoincidentSelect.onSetup = function(opts) {
   const planId = features.find(f => f.properties.vetro_id === feature.id)
     .properties.plan_id;
 
-  features.forEach(f => {
+  for(const f of features){
     if (
       f.properties.plan_id === planId &&
       f.geometry.type === "LineString" &&
       !f.layer.id.includes("_snap")
     ) {
-      const lineGeom = JSON.parse(f.properties.geojson_string);
+
+      let lineGeom = f.geometry;
+      if(typeof this._ctx.options.fetchSourceGeometry === "function"){
+
+        const [ lineSrcGeom, ptSrcGeom ] = await Promise.all([
+          this._ctx.options.fetchSourceGeometry(f),
+          this._ctx.options.fetchSourceGeometry({ properties: { vetro_id: state.initiallySelectedFeatureIds[0] }})
```

Comment:
```suggestion
          this._ctx.options.fetchSourceGeometry(f.properties.vetro_id),
          this._ctx.options.fetchSourceGeometry(state.initiallySelectedFeatureIds[0])
```
```suggestion
          this._ctx.options.fetchSourceGeometry(f),
          this._ctx.options.fetchSourceGeometry({ properties: { vetro_id: state.initiallySelectedFeatureIds[0] }})
```

Reply:
@lucasprograms I've made this update - not totally sure I agree with hardcoding a reference to VETRO here with `vetro_id`, but we can update this back if we want to make our fork of this library more public-friendly.

Reply:
One one hand we are just moving our reference to vetro_id to a different place (`{ properties: { vetro_id: state.initiallySelectedFeatureIds[0] }}` vs `f.properties.vetro_id`).

Though I absolutely agree it will be great if we can get this to a public friendly place! The divergence between our lib & the current mabbox-gl-draw is pretty wide at this point, unfortunately.